### PR TITLE
Connection hook for mysql

### PR DESCRIPTION
**Pull request by webfacebg**
This allows to execute an action on mysql connection. For example:

var sequelize = new Sequelize(
    process.env.MYSQL_DATABASE,
    process.env.MYSQL_USERNAME,
    process.env.MYSQL_PASSWORD, {
	host: process.env.MYSQL_HOST || 'localhst',
	onConnect: function (connection) {
	    connection.query("SET NAMES utf8;");
	}
    });

### CODE REVIEW

File: lib/dialects/mysql/connection-manager.js [hunk 1]
```diff
@@ -69,7 +69,8 @@ ConnectionManager.prototype.connect = function(config) {
 
         return;
       }
-
+	if (config.onConnect)
+	    config.onConnect(connection);
```
Comment (mickhansen) [inline lib/dialects/mysql/connection-manager.js]: Use brackets please. We mostly use single line if's for short-circuits.
There's also an indentation issue.

**Comment by webfacebg** [comment]
Brackets added :-)

**Comment by janmeier** [comment]
I like the idea, but don't think this is the best place to handle this. If you add it here instead https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/connection-manager.js#L228 it will apply to all dialects.

This also needs test and some documentation

I like the simplicity of just passing an option to the constructor, but on the other hand I think it would be better if this used the same API as the rest of our hooks system. What do you think @mickhansen ?

**Comment by webfacebg** [comment]
Please have a look at the last changes. I moved the "hook" to the abstract connection-manager. For some reason the onConnect parameter is not passed to the connection-manager, so I hat to put it inside the dialectParams.

File: lib/dialects/abstract/connection-manager.js [hunk 2]
```diff
@@ -73,7 +73,13 @@ ConnectionManager.prototype.initPools = function () {
       name: 'sequelize-connection',
       create: function(callback) {
         self.$connect(config).nodeify(function (err, connection) {
-          callback(err, connection); // For some reason this is needed, else generic-pool things err is a connection or some shit
+	  // Hook something on connection
+	  if (config.dialectOptions && config.dialectOptions.onConnect && typeof(config.dialectOptions.onConnect) === 'function') {
+	      // Do not forget to call the callback
+	      config.dialectOptions.onConnect(connection, err, callback);
```
Comment (janmeier) [inline lib/dialects/abstract/connection-manager.js]: I guess this should also only be called if `!err`

File: lib/dialects/abstract/connection-manager.js [hunk 3]
```diff
@@ -73,7 +73,13 @@ ConnectionManager.prototype.initPools = function () {
...
+	  } else {
+              callback(err, connection); // For some reason this is needed, else generic-pool things err is a connection or some shit
```
Comment (janmeier) [inline lib/dialects/abstract/connection-manager.js]: This comment can be removed now ;)

**Comment by janmeier** [comment]
Yeah, this looks good - there's some issues with the indentation though ;)  - should be 2 spaces

If you add API docs and tests this should be good to go

**Comment by mickhansen** [comment]
I agree it should probably have the same API as the other hooks, this does not appear to support promises?

**Comment by janmeier** [comment]
It doesn't support promises no, and I had the same though. But I guess its okay, since you're gonna have  to play around with callbacks anyway when you work directly with the connection

**Comment by mickhansen** [comment]
@janmeier Hmm, just not a huge fan of having a piece of the API be that much different, bound to give us headaches.

**Comment by janmeier** [comment]
Yeah, that's true. Ideally I'd like to use proper hooks for this, but thats a bit complicated too. If we add the hook to the  constructor (`Sequelize.addHook`) it will work fine, but you won't be able to add different hooks for different instances (would that be something someone wants??)

You could also add the hooks to the instance (`sequelize.addHook`), but that won't work if `pool.min` is set, because those connections would be initialized before the hook is addded.

Perhaps we can live with option 1, a global hook for all instances?

**Comment by webfacebg** [comment]
Added the error check, comments removed and indentation corrected. Could you please where/how  to add Docs and tests?

**Comment by mickhansen** [comment]
@janmeier well we already have the `hooks` option on `new Sequelize` which will add hooks to all models defined after that, wouldn't the same work for connection hooks i imagine?

**Comment by janmeier** [comment]
@mickhansen Yeah thats true - thats probably the best way to go :)

**Comment by janmeier** [comment]
@webfacebg Docs are in `docs/docs` - the ones in `/api` are auto generated from code comments.

For this I believe hand written docs would be fine.

As for tests, I believe `test/integration/sequelize.test` or `test/integration/dialects/abstract/` would be good places

**Comment by janmeier** [comment]
ping @webfacebg

**Comment by webfacebg** [comment]
В 11:54 -0700 на 08.10.2015 (чт), Jan Aagaard Meier написа:
> ping @webfacebg

Sorry. Just don't have the time to do it. It is on my list.

-- 
Anton Katsarov
WebFace — Your Web Face.
anton@webface.bg
http://www.webface.bg
Smolyan, Bulgaria
tel. +359 899 108 919

**Comment by janmeier** [comment]
Superceeded by https://github.com/sequelize/sequelize/pull/5688

**Pull request closed without merging**